source/common/http/filter_manager.h

```diff
@@ -112,6 +218,10 @@ struct ActiveStreamFilterBase : public virtual StreamFilterCallbacks,
   };
   FilterManager& parent_;
   IterationState iteration_state_;
+
+  Matcher::MatchTreeSharedPtr<HttpMatchingData> match_tree_;
```

Comment:
What would you say is the overhead of the new code if it's not being used? I mainly want to be cognizant of adding expensive computation / memory usage in the case where none of this is used. In an optimal world we would not allocate anything extra in the common case of no matching. Perhaps some more comments at the top of this file and/or block comments on top of the classes would help?

Reply:
I'll add some comments but the tl;dr is that since `evaluateMatchTreeWithNewData` does nothing unless `matching_data_` is set we should perform basically no additional work (besides a pointer check) when a match tree is not present. The only overhead would be storing two new pointers (which would both be nullptr) + a bit flag per filter wrapper. We could even drop this down to one pointer if we think this matters by adding a layer of indirection (so we just store one pointer, not two).